snap/snapenv/snapenv_test.go

```diff
@@ -186,25 +176,21 @@ func (s *HTestSuite) TestParallelInstallSnapRunSnapExecEnv(c *C) {
 
 		env := snapEnv(info)
 		c.Check(env, DeepEquals, map[string]string{
-			"HOME":                      fmt.Sprintf("%s/snap/snapname/42", usr.HomeDir),
-			"SNAP":                      fmt.Sprintf("%s/snapname/42", dirs.CoreSnapMountDir),
-			"SNAP_ARCH":                 arch.UbuntuArchitecture(),
-			"SNAP_COMMON":               "/var/snap/snapname/common",
-			"SNAP_DATA":                 "/var/snap/snapname/42",
-			"SNAP_LIBRARY_PATH":         "/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void",
-			"SNAP_NAME":                 "snapname",
-			"SNAP_INSTANCE":             fmt.Sprintf("%s/snapname_foo/42", dirs.CoreSnapMountDir),
-			"SNAP_INSTANCE_NAME":        "snapname_foo",
-			"SNAP_INSTANCE_COMMON":      "/var/snap/snapname_foo/common",
-			"SNAP_INSTANCE_DATA":        "/var/snap/snapname_foo/42",
-			"SNAP_REEXEC":               "",
-			"SNAP_REVISION":             "42",
-			"SNAP_USER_COMMON":          fmt.Sprintf("%s/snap/snapname/common", usr.HomeDir),
-			"SNAP_USER_DATA":            fmt.Sprintf("%s/snap/snapname/42", usr.HomeDir),
-			"SNAP_INSTANCE_USER_COMMON": fmt.Sprintf("%s/snap/snapname_foo/common", usr.HomeDir),
-			"SNAP_INSTANCE_USER_DATA":   fmt.Sprintf("%s/snap/snapname_foo/42", usr.HomeDir),
-			"SNAP_VERSION":              "1.0",
-			"XDG_RUNTIME_DIR":           fmt.Sprintf("/run/user/%d/snap.snapname_foo", sys.Geteuid()),
+			"HOME":               fmt.Sprintf("%s/snap/snapname_foo/42", usr.HomeDir),
+			"SNAP":               fmt.Sprintf("%s/snapname/42", dirs.CoreSnapMountDir),
+			"SNAP_ARCH":          arch.UbuntuArchitecture(),
+			"SNAP_COMMON":        "/var/snap/snapname/common",
+			"SNAP_DATA":          "/var/snap/snapname/42",
```

Comment:
Is this correct? Or should this be `/var/snap/snapname_foo/42` ?

Reply:
Yes, SNAP, SNAP_COMMON and SNAP_DATA will be handled by snap-update-ns to make them appear as if it's an instance-key-less snap.

Reply:
Could we add a comment somewhere in the code about this? Just to ensure my future forgetful self remembers this?

Reply:
There is one here https://github.com/snapcore/snapd/pull/5735/files/9ded005f63e58159ff9d883b7068744ed7fd7cc0#diff-2207380360ab8eacdc1bf498f3eaaf35R128 but I'll add another one in the test too.